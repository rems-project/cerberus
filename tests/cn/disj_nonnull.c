int x = 1;
int y = 2;

void globals()
/*@ accesses x; accesses y; @*/
{

  /*@ assert((u64) &x != (u64) &y); @*/

  /*@ assert((u64)&x < (u64)&x + 4u64); @*/
  /*@ assert((u64)&x < MAXu64() - 4u64); @*/

  /*@ assert((u64)&y < MAXu64() - 4u64); @*/
  /*@ assert((u64)&y < (u64)&y + 4u64); @*/

  /*@ assert((u64)&x < (u64)&y || (u64)&x > (u64)&y); @*/
  /*@ assert((u64)&x + 4u64 <= (u64)&y || (u64)&y + 4u64 <= (u64)&x); @*/

}

int main()
{
    int p = 1;
    int q = 2;

  /*@ assert((u64) &p != (u64) &q); @*/

  /*@ assert((u64)&p < (u64)&p + 4u64); @*/
  /*@ assert((u64)&p < MAXu64() - 4u64); @*/

  /*@ assert((u64)&q < MAXu64() - 4u64); @*/
  /*@ assert((u64)&q < (u64)&q + 4u64); @*/

  /*@ assert((u64)&p < (u64)&q || (u64)&p > (u64)&q); @*/
  /*@ assert((u64)&p + 4u64 <= (u64)&q || (u64)&q + 4u64 <= (u64)&p); @*/

}
