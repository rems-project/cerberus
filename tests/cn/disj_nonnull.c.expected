return code: 0
[1/2]: globals -- pass
[2/2]: main -- pass
